src/zitadel.rs

```diff
@@ -430,3 +465,9 @@ impl From<User> for ImportHumanUserRequest {
 		}
 	}
 }
+
+impl Display for User {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		write!(f, "email={}", &self.email)
```

Comment:
this user string representation is confusing, I'd suggest to `format!` where we need it in-place, if we need it in several places, we can make a function with descriptive name like `user_as_ldapish_string`

Reply:
I mostly wanted a canonical way of displaying the user in log messages; Before this I used `uid`, which turned out to be binary in AD, and flipping every invocation of `user.ldap_id` to `user.email` was annoying, and pointed to needing a `Display` implementation.

As such, I'm not really sure we need to complicate this. Isn't `Display` intended exactly for this, printing user-facing values?

Reply:
And then we also have a bunch of other fields in `User` that we just simply ignore here, and it's a bit meh. I think if we want to display ids we should do `"user {user.id}"`, it's also a bit more explicit in the code, making it clear that only id (or email) is displayed, not the whole user. 

As for "what should we use as id, `ldap_id` or `email`" we can have some method in `User` like `identifier` or `display_identifier` or `printable_identifier` or whatever.

Reply:
> And then we also have a bunch of other fields in User that we just simply ignore here

Yeah, if we want a representation that shows all the object's details that's what the `Debug` impl is for.

I'd really rather not have each log potentially log a different attribute as the core identifier, or hardcode it to be represented by a specific attribute in case either that or the exact desired way of representing it changes (we may want to switch to `dn` if group sync ends up being implemented, for example, which is a composite of attributes).

If this is a blocker, we can implement a separate `fmt_for_log`, though I maintain that `Display` at least appears to be the intended trait for that.

Reply:
> Yeah, if we want a representation that shows all the object's details that's what the `Debug` impl is for.

but it's also for debugging, and generally "I don't care how exactly it looks, derive me something I can dump to stdout during debugging".

Can you find an example of a type with `Display` that omits some of the data (most of the data here). I can't think of a single example of a type that would implement `Display` partially, especially for just one field. If I see `user.to_string()` I expect to get user representation in text, `Debug`ish or `Display`ish. Getting just identifier seems unobvious and confusing

> If this is a blocker, we can implement a separate fmt_for_log

I'd suggest to go with this approach, however I'd call it something like `identifier` (`dn` is also identifier)

P.S. The situation in haskell is even [worse](https://harry.garrood.me/blog/down-with-show-part-2/), there's only one `Show` typeclass/trait, which is used for debuging and "displaying", and there's no clear definition of what "displaying" is, as you can have many options: multiline/singleline, compact/pretty/formatted, full/redacted, colored/colorless, syntaxA/syntaxB etc

Reply:
I ended up going with `.log_name()`, that seemed the most simple term for this.